diff -r f24756eda3d6 src/windows/native/java/net/Inet4AddressImpl.c
--- a/src/windows/native/java/net/Inet4AddressImpl.c	Tue Jun 16 09:46:02 2020 +0200
+++ b/src/windows/native/java/net/Inet4AddressImpl.c	Thu Jun 18 09:46:56 2020 -0700
@@ -484,7 +484,7 @@
     ReplyBuffer = (VOID*) malloc(ReplySize);
     if (ReplyBuffer == NULL) {
         IcmpCloseHandle(hIcmpFile);
-        NET_ThrowNew(env, WSAGetLastError(), "Unable to allocate memory");
+        NET_ThrowNew(env, -1, "Unable to allocate memory");
         return JNI_FALSE;
     }
 
diff -r f24756eda3d6 src/windows/native/java/net/Inet6AddressImpl.c
--- a/src/windows/native/java/net/Inet6AddressImpl.c	Tue Jun 16 09:46:02 2020 +0200
+++ b/src/windows/native/java/net/Inet6AddressImpl.c	Thu Jun 18 09:46:56 2020 -0700
@@ -460,7 +460,7 @@
     ReplyBuffer = (VOID*) malloc(ReplySize);
     if (ReplyBuffer == NULL) {
         IcmpCloseHandle(hIcmpFile);
-        NET_ThrowNew(env, WSAGetLastError(), "Unable to allocate memory");
+        NET_ThrowNew(env, -1, "Unable to allocate memory");
         return JNI_FALSE;
     }
 
diff -r f24756eda3d6 src/windows/native/java/net/SocketInputStream.c
--- a/src/windows/native/java/net/SocketInputStream.c	Tue Jun 16 09:46:02 2020 +0200
+++ b/src/windows/native/java/net/SocketInputStream.c	Thu Jun 18 09:46:56 2020 -0700
@@ -134,12 +134,13 @@
         (*env)->SetByteArrayRegion(env, data, off, nread, (jbyte *)bufP);
     } else {
         if (nread < 0) {
+            int err = WSAGetLastError();
             // Check if the socket has been closed since we last checked.
             // This could be a reason for recv failing.
             if ((*env)->GetIntField(env, fdObj, IO_fd_fdID) == -1) {
                 NET_ThrowSocketException(env, "Socket closed");
             } else {
-                switch (WSAGetLastError()) {
+                switch (err) {
                     case WSAEINTR:
                         JNU_ThrowByName(env, JNU_JAVANETPKG "SocketException",
                             "socket closed");
